tools/StandardAnchorTags/TocSectionNumberBuilder.cs

```diff
@@ -53,12 +53,24 @@ public TocSectionNumberBuilder(string pathFromToolToStandard, bool dryRun)
         /// <summary>
         /// Add the front matter entries to the TOC
         /// </summary>
-        /// <param name="entries">A params array of tuples that contains the link text and the file names.</param>
-        public void AddFrontMatterTocEntries(params (string linkText, string fileName)[] entries)
+        /// <param name="fileName">The front matter file name.</param>
+        /// <remarks>
+        /// The front matter entries don't add section numbers in the headers.
+        /// Otherwise, this is the same logic for the main file.
+        /// </remarks>
+        public async Task AddFrontMatterTocEntries(string fileName)
         {
-            foreach (var entry in entries)
+            string? line;
+            using var stream = new StreamReader($"{PathToStandardFiles}/{fileName}");
+            while ((line = await stream.ReadLineAsync()) != null)
             {
-                tocContent.AppendLine($"- [{entry.linkText}]({entry.fileName})");
+                if (line.StartsWith("# "))
+                {
+                    var linkText = line[2..];
+                    tocContent.AppendLine($"- [{linkText}]({fileName})");
+                    // Done: return.
+                    return;
+                }
```

Comment:
Do we want to throw an exception if we get to the end of this method? It means we haven't found a heading, which is presumably a problem.

Also, can we not just read the first line of the file and assert that it starts with `# `? (Again, if it doesn't that's presumably something wrong...)

Reply:
Both good suggestions. (I thought about the 2nd, but then wondered if we might ever add comments above the H1. I tend to agree that we won't do that).